site/content/integrate/apps/deployment/_index.md

```diff
@@ -52,40 +49,128 @@ For a go app the manifest snippet would look like this:
+
+Run the following commands to test and validate your AWS configuration and permissions.
 
 ```bash
-go install github.com/mattermost/mattermost-plugin-apps/cmd/appsctl@latest
-appsctl provision app /PATH/TO/YOUR/APP/BUNDLE
+go run ./cmd/appsctl aws test lambda
+go run ./cmd/appsctl aws test s3
 ```
 
-It reads appropriate AWS credentials from environment variables:
+#### Optional: Provision the hello-lambda app
 
-`APPS_PROVISION_AWS_ACCESS_KEY`
+A test command is also available if you would like to test provisioning the hello-lambda test app. This will add a manifest to the S3 bucket and add a corresponding lambda function to AWS.
 
-`APPS_PROVISION_AWS_SECRET_KEY`
+Build hello-lambda bundle
 
-We need an app bundle to provision an app. The bundle might be provisioned from the local disk, from S3 (not implemented yet), or from a URL (not implemented yet). Provisioning consists of three parts:
+- `cd ./examples/go/hello-lambda && make dist; cd -`
+- `go run ./cmd/appsctl aws test provision ./examples/go/hello-lambda/dist/bundle.zip`
 
-1. Creating the lambda functions with appropriate policies.
-2. Storing static assets in the dedicated S3 bucket.
-3. Storing the app’s manifest file in the same dedicated S3 bucket.
+### Deploying a custom app to AWS
+
+After setting up your Mattermost instance, AWS key, and S3 bucket you can now provision your app using `appsctl`. Note that `appsctl` commands are run in the `mattermost-plugin-apps` repo.
+
+Install appsctl  
+
+- `go install github.com/mattermost/mattermost-plugin-apps/cmd/appsctl@latest`
 
-AWS Lambda functions have semantic names, which means that a function described in the `manifest.json` file translates to AWS as `$appID_$appVersion_$functionName` to avoid collisions with other apps' or other versions' functions. And **appsctl** provisions lambda functions using this name. For example the name of a `servicenow` app's lambda function might be `com-mattermost-servicenow_0-1-0_go-function`. You don't need to worry about the AWS Lambda function names, as the Apps Plugin takes care of it. The dedicated S3 bucket name is stored in the environment variable:
+Provision your app  
 
-`MM_APPS_S3_BUCKET`
+- `appsctl provision app /PATH/TO/YOUR/APP/BUNDLE.zip`
+
+The provision command output will show the `/apps` installation to use in Mattermost.
+
+### Provisioned app details
+
+AWS Lambda functions have semantic names, which means that a function described in the `manifest.json` file translates to AWS as `$appID_$appVersion_$functionName` to avoid collisions with other apps' or other versions' functions. And **appsctl** provisions lambda functions using this name. For example the name of a `servicenow` app's lambda function might be `com-mattermost-servicenow_0-1-0_go-function`. You don't need to worry about the AWS Lambda function names, as the Apps Plugin takes care of it. The dedicated S3 bucket name is stored in the environment variable: `MM_APPS_S3_BUCKET`.
```

Comment:
```suggestion
AWS Lambda functions have semantic names, which means that a function described in the `manifest.json` file translates to AWS as `$appID_$appVersion_$functionName` to avoid collisions with other apps' or other versions' functions. And **appsctl** provisions lambda functions using this name. For example the name of a `servicenow` app's lambda function might be `com-mattermost-servicenow_0-1-0_go-function`. You don't need to worry about the AWS Lambda function names, as **appsctl** care of it. The dedicated S3 bucket name is stored in the environment variable: `MM_APPS_S3_BUCKET`.
```